google/cloud/firestore_v1/base_document.py

```diff
@@ -355,7 +358,7 @@ def get(self, field_path) -> Any:
         nested_data = field_path_module.get_nested_value(field_path, self._data)
         return copy.deepcopy(nested_data)
 
-    def to_dict(self) -> Any:
+    def to_dict(self) -> Optional[Dict[str, Any]]:
```

Comment:
Likewise, this would be clearer as:
```suggestion
    def to_dict(self) -> Union[Dict[str, Any], None]:
```